build.rs

```diff
@@ -39,6 +39,10 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
 
     make.env_remove("PROFILE").arg("-C").arg(&out_dir).arg("build");
 
+    if env::var("TARGET").unwrap().ends_with("-musl") {
+        make.env("CC", "musl-gcc");
```

Comment:
Could we change this so it respects `CC` if set in the environment? (i.e. only hardcode CC to `musl-gcc` if `CC` is not set, and we're on a target ending in `-musl)

(per separate discussion with @msepga, who is better versed at Rust than I am 😄)